### PR TITLE
Connection singleton

### PR DESCRIPTION
**Pull request by eckelj**
A connection singleton got introduced to avoid connection issues and unstables states and connection configurations.

## Checklist

- [X] version number increased
- [X] CHANGELOG.md updated
- [X] unit test added/adjusted

### CODE REVIEW

File: planetmint/backend/schema.py [hunk 1]
```diff
@@ -134,7 +134,12 @@ def init_database(connection=None, dbname=None):
             configuration.
     """
 
-    connection = connection or connect()
+    connection = connection or Connection()
+    print(
```
Comment (LaurentMontBlanc) [inline planetmint/backend/schema.py:138]: remove before merging

File: tests/tendermint/test_fastquery.py [hunk 2]
```diff
@@ -116,7 +116,11 @@ def test_outputs_query_key_order(b, user_pk, user_sk, user2_pk, user2_sk):
     assert len(outputs) == 1
 
     # clean the transaction, metdata and asset collection
-    connection = connect()
+    # conn = connect()
```
Comment (LaurentMontBlanc) [inline tests/tendermint/test_fastquery.py:119]: remove comments before merging

**Review by LaurentMontBlanc — CHANGES_REQUESTED** [review]
Looks good to me. Can be merged as soon as comments are resolved and abci test are green 👍🏼

File: tests/tendermint/test_fastquery.py (hunk 2 above)
Comment (eckelj) [reply tests/tendermint/test_fastquery.py]: thx for mentioning. just did so!

File: tests/tendermint/test_lib.py [hunk 3]
```diff
@@ -5,6 +5,12 @@
 
 
 import os
+
+# from unittest.mock import patch
+# from planetmint.transactions.types.assets.create import Create
+# from planetmint.transactions.types.assets.transfer import Transfer
+# from hashlib import sha3_256
+
```
Comment (LaurentMontBlanc) [inline tests/tendermint/test_lib.py:13]: remove unused lines

File: tests/web/test_outputs.py [hunk 4]
```diff
@@ -5,6 +5,11 @@
 
 import pytest
 
+# from unittest.mock import MagicMock, patch
+# from planetmint.transactions.common import crypto
+# from planetmint.transactions.types.assets.create import Create
+# from planetmint.transactions.types.assets.transfer import Transfer
+
```
Comment (LaurentMontBlanc) [inline tests/web/test_outputs.py:12]: remove unused lines

File: tests/web/test_transactions.py [hunk 5]
```diff
@@ -11,6 +11,11 @@
 from cryptoconditions import Ed25519Sha256
 from ipld import multihash, marshal
 from hashlib import sha3_256
+
+# from planetmint.transactions.common import crypto
+# from planetmint.transactions.types.assets.create import Create
+# from planetmint.transactions.types.assets.transfer import Transfer
+# from planetmint.transactions.common.transaction_mode_types import (
```
Comment (LaurentMontBlanc) [inline tests/web/test_transactions.py:18]: remove unused lines

File: docs/root/source/connecting/http-samples/index-response.http [hunk 6]
```diff
@@ -15,7 +15,7 @@ Content-Type: application/json
       "validators": "/api/v1/validators"
     }
   },
-  "docs": "https://docs.planetmint.io/projects/server/en/v1.0.1/",
+  "docs": "https://docs.planetmint.io/projects/server/en/v1.2.2/",
   "software": "Planetmint",
-  "version": "1.0.1"
+  "version": "1.2.2"
```
Comment (LaurentMontBlanc) [inline docs/root/source/connecting/http-samples/index-response.http:20]: increase to 1.3.1

File: docs/root/source/connecting/http-samples/api-index-response.http [hunk 7]
```diff
@@ -4,7 +4,7 @@ Content-Type: application/json
 {
   "assets": "/assets/",
   "blocks": "/blocks/",
-  "docs": "https://docs.planetmint.io/projects/server/en/v1.0.1/http-client-server-api.html",
+  "docs": "https://docs.planetmint.io/projects/server/en/v1.2.2/http-client-server-api.html",
```
Comment (LaurentMontBlanc) [inline docs/root/source/connecting/http-samples/api-index-response.http:7]: increase to 1.3.1

**Review by LaurentMontBlanc — APPROVED** [review]
looks good to me, commented on some minor stuff. is mergeable afterwards

File: docs/root/source/connecting/http-samples/api-index-response.http (hunk 7 above)
Comment (eckelj) [reply docs/root/source/connecting/http-samples/api-index-response.http]: thx! done!

File: tests/tendermint/test_lib.py (hunk 3 above)
Comment (eckelj) [reply tests/tendermint/test_lib.py]: thx! done!

**Pull request merged**